lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1553,18 +1553,14 @@ private boolean closeImpl() {
             return true;
         }
 
-        /* if connected, send an explicit close message and await response */
-        boolean isConnected = currentState.state == ConnectionState.connected;
-        if(isConnected) {
-            try {
-                Log.v(TAG, "Requesting connection close");
-                transport.send(new ProtocolMessage(ProtocolMessage.Action.close));
-                return false;
-            } catch (AblyException e) {
-                /* we're closing, and the attempt to send the CLOSE message failed;
-                 * continue, because we're not going to reinstate the transport
-                 * just to send a CLOSE message */
-            }
+        try {
+            Log.v(TAG, "Requesting connection close");
```

Comment:
Should we keep the check here, but change it to `connected || closing` - we probably don't want to be sending messages if we're in any other state?

Reply:
that's a fair point, i did consider doing that - tbh i think it's safer to just send the close message indiscriminately just as long as there's a transport available to send it on. can't think of a situation where we have a transport and we don't want to send a close message when closing

Reply:
Fair :)